src/pages/product-details/product-details-page.js

```diff
@@ -0,0 +1,250 @@
+
+  const uniqueSizes = useMemo(
+    () => [
+      ...new Set(
+        options
+          ? options.map(({ size: { available, name } }) => available && name)
+          : null
+      )
+    ],
+    [options]
+  );
+
+  const uniqueBottomMaterials = useMemo(
+    () => [
+      ...new Set(
+        options
+          ? options.map(({ bottomMaterial: item }) =>
+            item && item.available ? item.name[1].value : null
+          )
+          : null
+      )
+    ],
+    [options]
+  );
+
+  const uniqueAdditions = useMemo(
+    () => [
+      ...new Set(
+        options
+          ? options
+            .filter(({ additions }) => additions.length > 0)
+            .map(
+              ({ additions: [{ available, name }] }) =>
+                available && name[1].value
+            )
+          : null
+      )
+    ],
+    [options]
+  );
+
+  const sizes = useMemo(
+    () =>
+      uniqueSizes.map(
+        (item) => options.find(({ size: { name } }) => item === name).size
+      ),
+    [uniqueSizes, options]
+  );
+
+  const bottomMaterials = useMemo(
+    () =>
+      uniqueBottomMaterials[0]
+        ? uniqueBottomMaterials.map(
+          (item) =>
+            options.find(
+              ({ bottomMaterial: { name } }) => item === name[1].value
+            ).bottomMaterial
+        )
+        : null,
+    [uniqueBottomMaterials, options]
+  );
+
+  const additions = useMemo(
+    () =>
+      uniqueAdditions.map(
+        (item) =>
+          options
+            .filter(({ additions }) => additions.length > 0)
+            .find(({ additions: [{ name }] }) => item === name[1].value)
+            .additions[0]
+      ),
+    [uniqueAdditions, options]
+  );
+
+  const bottomNameToSend = useMemo(
+    () =>
+      bottomMaterials
+        ? bottomMaterials.find(({ name }) => name[1].value === bagBottom)
+        : null,
+    [bagBottom, bottomMaterials]
+  );
+
+  const additionsNameToSend = useMemo(
+    () => (additions.length >= 1 ? additions[0].name : null),
+    [additions]
+  );
+
+  const productToSend = {
+    _id,
+    name,
+    images,
+    selectedSize,
+    bagBottom: bottomNameToSend,
+    sidePocket: {
+      isSelected: sidePocket,
+      name: additionsNameToSend
+    },
+    totalPrice: currentPrice,
+    quantity: 1,
+    productUrl
+  };
+
+  const checkSize = () => {
+    if (sizes[0].name && !selectedSize) {
+      setError(true);
+      return false;
+    }
+    return true;
+  };
+
+  const handleSizeChange = (event) => {
+    const { textContent } = event.target;
+
+    const oldPrice = selectedSize
+      ? sizes.find(({ name }) => name === selectedSize).additionalPrice[0].value
+      : 0;
+
+    const { additionalPrice, volumeInLiters, weightInKg } = sizes.find(
+      ({ name }) => name === textContent
+    );
+
+    setSize(textContent);
+    setPrice((price) => price - oldPrice + additionalPrice[0].value);
+    setDimensions({ volumeInLiters, weightInKg });
+
+    if (error) {
```

Comment:
Because if user is trying to add product to cart or checkout and don't select size, it throws an error, so after that if user select size i need to remove that error from UI, so i `setError` to `false`